actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -95,9 +96,23 @@ def clear!
             @url_helpers_module.send  :undef_method, helper
           end
 
+          @custom_helpers.each do |helper|
+            path_name = :"#{helper}_path"
+            url_name = :"#{helper}_url"
+
+            if @path_helpers_module.method_defined?(path_name)
+              @path_helpers_module.send :undef_method, path_name
```

Comment:
Given the fact that we should allow people to access inherited methods if they include url helpers and then change a conflicting route name then probably best to switch to `remove_method`